include/deal.II/grid/manifold.h

```diff
@@ -273,13 +273,36 @@ class Manifold : public Subscriptor
   /// @{
 
   /**
+   * Return an intermediate point between two given
+   * points. Overloading this function allows the default recursive
+   * implementation of the method get_new_point() that takes a
```

Comment:
I think the term "recursive" is not quite correct here since there is no function that calls itself. Maybe use the term "pairwise reduction" we use in the paper?